popup.js

```diff
@@ -12,23 +12,23 @@ const AGENCY_IDS = {
 };
 
 function debounce(fn, duration) {
-  let timeout = null;
+  var timeout = null;
 
   return function () {
-    const args = arguments;
+    var args = arguments;
 
     if (timeout) {
       clearTimeout(timeout);
     }
 
-    timeout = setTimeout(() => {
+    timeout = setTimeout(function(){
       timeout = null;
       fn.apply(this, args);
```

Comment:
Argh, forgot. Feel free to take over my branch, otherwise I'll get to it on the weekend -- busy the next few days